src/lib/entities/audio/model/audio-instance.ts

```diff
@@ -1,93 +1,24 @@
-import { writable, derived } from 'svelte/store';
+import { writable } from 'svelte/store';
 
-export const audio = writable<HTMLAudioElement | undefined>(undefined);
+let audio: HTMLAudioElement | undefined;
+export const currentTime = writable(0);
+export const duration = writable(NaN);
+export const playbackRate = writable(1);
+export const paused = writable(false);
 
-/** Tracks the duration of the audio stream that is currently loaded in the `audio` store. */
-export const audioDuration = derived<typeof audio, number>(
-  audio,
-  ($audio, set) => {
-    if ($audio === undefined) {
-      set(NaN);
-      return;
-    }
-
-    function watchDuration(event: Event) {
-      set((event.target as HTMLAudioElement).duration);
-    }
-
-    $audio.addEventListener('durationchange', watchDuration);
-
-    return () => $audio.removeEventListener('durationchange', watchDuration);
-  },
-  NaN
-);
-
-/** Tracks the playback position in the audio stream that is currently loaded in the `audio` store. */
-export const audioPosition = derived<typeof audio, number>(
-  audio,
-  ($audio, set) => {
-    if ($audio === undefined) {
-      set(0);
-      return;
-    }
-
-    function watchProgress(event: Event) {
-      set((event.target as HTMLAudioElement).currentTime);
-    }
-
-    $audio.addEventListener('timeupdate', watchProgress);
-
-    return () => $audio.removeEventListener('timeupdate', watchProgress);
-  },
-  0
-);
-
-export const play = (src?: string) => {
-  audio.update(($audio) => {
-    if ($audio === undefined) {
-      return;
-    }
-
-    if (src) {
-      $audio.src = src;
-    }
-
-    $audio.play();
-    return $audio;
-  });
+export const setAudio = (value: HTMLAudioElement) => {
+  audio = value;
 };
 
-export const pause = () => {
-  audio.update(($audio) => {
-    if ($audio === undefined) {
-      return;
-    }
-
-    $audio.pause();
-    return $audio;
-  });
+export const play = (src?: string) => {
+  if (audio && src) audio.src = src;
+  paused.set(false);
 };
 
-export const seek = (timeInSeconds: number) => {
-  audio.update(($audio) => {
-    if ($audio === undefined) {
-      return;
-    }
+export const pause = () => paused.set(true);
 
-    $audio.currentTime = timeInSeconds;
-    $audio.play();
-    return $audio;
-  });
-};
+export const seek = (timeInSeconds: number) => currentTime.set(timeInSeconds);
 
 export const move = (deltaInSeconds: number) => {
-  audio.update(($audio) => {
-    if ($audio === undefined) {
-      return;
-    }
-
-    $audio.currentTime += deltaInSeconds;
-    $audio.play();
-    return $audio;
-  });
+  if (audio) currentTime.set(audio.currentTime + deltaInSeconds);
```

Comment:
suggestion: Writable stores have an `update` method that receive the previous value of the store. This could help eliminate the dependency on `audio` here